config/src/inline/mod.rs

```diff
@@ -0,0 +1,81 @@
+mod conf_parser;
+pub use conf_parser::{
+    parse_config_bool, parse_config_u32, validate_profiles, InlineConfigParser,
+    InlineConfigParserError,
+};
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+
+mod natspec;
+pub use natspec::NatSpec;
+
+use crate::Config;
+
+pub const INLINE_CONFIG_FUZZ_KEY: &str = "fuzz";
+pub const INLINE_CONFIG_INVARIANT_KEY: &str = "invariant";
+const INLINE_CONFIG_PREFIX: &str = "forge-config";
+
+static INLINE_CONFIG_PREFIX_SELECTED_PROFILE: Lazy<String> = Lazy::new(|| {
+    let selected_profile = Config::selected_profile().to_string();
+    format!("{INLINE_CONFIG_PREFIX}:{selected_profile}.")
+});
+
+/// Wrapper error struct that catches config parsing
+/// errors [`InlineConfigParserError`], enriching them with context information
+/// reporting the misconfigured line.
+#[derive(thiserror::Error, Debug)]
+#[error("Inline config Error detected at {line} {source}")]
+pub struct InlineConfigError {
+    /// Specifies the misconfigured line. This is something of the form
+    /// `dir/TestContract.t.sol:FuzzContract:10:12:111`
+    pub line: String,
+    /// The inner error
+    pub source: InlineConfigParserError,
```

Comment:
Right now the error looks like this:

```
Inline config Error detected at test/Counter.t.sol:CounterTest:testSetNumber1:"288:82:20"
```

Not sure if useful, but I think @mattsse added an `offset_to_line` to the forge fmt error messages which may useful to get a line number here

Reply:
Related, it looks like the error is actually duplicated, once in red, and again in the "context" section. Is that intentional?

<img width="969" alt="image" src="https://user-images.githubusercontent.com/17163988/236897456-83abac64-571b-4c77-929b-ef6b237cb1f5.png">